<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <!-- If multiple robots set multiple=true and define the namespace -->
    <arg name="robot_name" default="pa10"/>
    <arg name="multiple" default="true"/>
    <arg name="multi" value="multiple:=$(arg multiple) namespace:=$(arg robot_name)"/>
    
  <!-- The semantic description that corresponds to the URDF -->
  <param name="/$(arg robot_name)/robot_description_semantic" textfile="$(find pa10_moveit_config)/config/pa10.srdf" />
  <rosparam command="load" file="$(find pa10_moveit_config)/config/kinematics.yaml" ns="/$(arg robot_name)"/>
  
  	<rosparam file="$(find pa10_hardware_interface)/config/real_controllers.yaml" command="load" ns="/$(arg robot_name)"/>
	
	<param name="/$(arg robot_name)/robot_description" command="$(find xacro)/xacro --inorder '$(find pa10_description)/urdf/pa10_csl.urdf.xacro' $(arg multi)"/>

  <!-- load the controllers -->
	<node name="pa10_hardware_interface" pkg="pa10_hardware_interface" type="pa10_hardware_interface_node" ns="/$(arg robot_name)"/>

  <node name="pa10_trajectory_controller" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" args="
      --namespace=/$(arg robot_name)
      joint_state_controller 
      arm_controller
      gripper_controller "/>
      
      
      <node pkg="tf" type="static_transform_publisher" name="$(arg robot_name)_bc" args="0.0 0.0 0.0 0.0 0.0 0.0 /world $(arg robot_name)/world 100"/>
      
  <!-- Robot state publisher -->
  <node pkg="robot_state_publisher" type="state_publisher" name="$(arg robot_name)_state_publisher">
            <param name="publish_frequency" type="double" value="50.0"/>
            <param name="tf_prefix" type="string" value="$(arg robot_name)"/>
            <remap from="robot_description" to="$(arg robot_name)/robot_description" />
            <remap from="joint_states" to="$(arg robot_name)/joint_states" />
  </node>
  
  <!-- Use MoveIt -->
        <group ns="$(arg robot_name)">
            <!-- Launch MoveIt -->
            <include file="$(find pa10_moveit_config)/launch/pa10_moveit.launch">
                <arg name="rviz" value="true" />
                <arg name="camera" value="false"/>
            </include>
        </group>

</launch>

